internal/models/one_time_token.go

```diff
@@ -243,17 +243,25 @@ func FindUserByEmailChangeCurrentAndAudience(tx *storage.Connection, email, toke
 
 // FindUserByEmailChangeNewAndAudience finds a user with the matching email change and audience.
 func FindUserByEmailChangeNewAndAudience(tx *storage.Connection, email, token, aud string) (*User, error) {
+	var ottFound bool
 	ott, err := FindOneTimeToken(tx, token, EmailChangeTokenNew)
 	if err != nil && !IsNotFoundError(err) {
 		return nil, err
+	} else if err == nil {
+		ottFound = true
```

Comment:
why don't we check for `ott != nil` instead?

Reply:
works